Add ADR for DI

InAnYan opened the pull request:
### **User description**
Closes https://github.com/JabRef/jabref-issue-melting-pot/issues/590

Just added an ADR.

### Steps to test

No steps to test. Just check the contents.

### Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] I manually tested my changes in running JabRef (always required)
- [/] I added JUnit tests for changes (if applicable)
- [/] I added screenshots in the PR description (if change is visible to the user)
- [/] I described the change in `CHANGELOG.md` in a way that is understandable for the average user (if change is visible to the user)
- [x] I checked the [user documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request updating file(s) in <https://github.com/JabRef/user-documentation/tree/main/en>.


___

### **PR Type**
Documentation


___

### **Description**
- Adds ADR 0055 documenting dependency injection strategy

- Defines hybrid approach: DI framework for JavaFX views, constructor-based DI for core logic

- Explains rationale balancing JavaFX constraints with architectural clarity

- Documents decision drivers, options considered, and verification methods


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["DI Strategy Decision"] --> B["JavaFX Views<br/>DI Framework"]
  A --> C["Core Logic<br/>Constructor-based DI"]
  B --> D["Empty constructors<br/>Field injection"]
  C --> E["Explicit dependencies<br/>Testable"]
```



<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>0055-dependency-injection-approach.md</strong><dd><code>ADR for hybrid dependency injection approach</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/decisions/0055-dependency-injection-approach.md

<ul><li>New ADR document establishing hybrid dependency injection strategy<br> <li> Defines use of DI framework for JavaFX views with empty constructors<br> <li> Specifies constructor-based DI for core logic layer<br> <li> Documents decision drivers, considered options, consequences, and <br>verification methods</ul>


</details>


  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14466/files#diff-df9100de4a263e5da3e27da789ed34f921de20b68fd5fbf06ef5a870228750a1">+67/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tbody></table>

</details>

___

InAnYan committed:
feat(docs): add ADR for DI

jabref-machine commented:
You modified Markdown (`*.md`) files and did not meet JabRef's rules for consistently formatted Markdown files. To ensure consistent styling, we have [markdown-lint](https://github.com/DavidAnson/markdownlint) in place. [Markdown lint's rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#rules) help to keep our Markdown files consistent within this repository and consistent with the Markdown files outside here.

You can check the detailed error output by navigating to your pull request, selecting the tab "Checks", section "Source Code Tests" (on the left), subsection "Markdown".


<!-- ghprcomment
Source Code Tests
Markdown
-->

InAnYan committed:
small fixes

InAnYan committed:
Merge branch 'main' into feat/di-adr

jabref-machine commented:
You modified Markdown (`*.md`) files and did not meet JabRef's rules for consistently formatted Markdown files. To ensure consistent styling, we have [markdown-lint](https://github.com/DavidAnson/markdownlint) in place. [Markdown lint's rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#rules) help to keep our Markdown files consistent within this repository and consistent with the Markdown files outside here.

You can check the detailed error output by navigating to your pull request, selecting the tab "Checks", section "Source Code Tests" (on the left), subsection "Markdown".


<!-- ghprcomment
Source Code Tests
Markdown
-->

Siedlerchr committed:
Merge branch 'main' into feat/di-adr

InAnYan committed:
Refactor dependency injection decision document

InAnYan committed:
Refine context for dependency injection strategy

InAnYan committed:
Refine dependency injection approach and analysis

Clarified the approach to dependency injection by specifying the use of a DI framework for JavaFX view models and constructor-based DI for core code. Added analysis of the choice highlighting pros and cons.

InAnYan committed:
Add header to dependency injection decision record

Added a header to the dependency injection decision record.

InAnYan committed:
Merge branch 'main' into feat/di-adr

InAnYan committed:
Update nav_order for Dependency Injection decision

jabref-machine commented:
You modified Markdown (`*.md`) files and did not meet JabRef's rules for consistently formatted Markdown files. To ensure consistent styling, we have [markdown-lint](https://github.com/DavidAnson/markdownlint) in place. [Markdown lint's rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#rules) help to keep our Markdown files consistent within this repository and consistent with the Markdown files outside here.

You can check the detailed error output by navigating to your pull request, selecting the tab "Checks", section "Source Code Tests" (on the left), subsection "Markdown".


<!-- ghprcomment
Source Code Tests
Markdown
-->

InAnYan committed:
Fix grammatical issues in dependency injection decision

Corrected grammatical errors in the document.

InAnYan committed:
Fix formatting in dependency injection decision document

InAnYan committed:
Clarify afterburnerfx relation to ServiceLocator pattern

Added note on the similarity of afterburnerfx to the ServiceLocator pattern.

InAnYan committed:
Remove reference to afterburnerfx and ServiceLocator

Removed mention of 'afterburnerfx' and its similarity to the 'ServiceLocator' pattern.

InAnYan commented:
I think the pr is good to go

The only problem is the koppor's comment

calixtus commented:
I think your rewording sounds fine. Integrate it. Then merge

InAnYan committed:
Refine dependency injection decision details

jabref-machine commented:
JUnit tests of `jabsrv` are failing. You can see which checks are failing by locating the box "Some checks were not successful" on the pull request page. To see the test output, locate "Source Code Tests / Unit tests (pull_request)" and click on it.

You can then run these tests in IntelliJ to reproduce the failing tests locally. We offer a quick test running howto in the section [Final build system checks](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.html#final-build-system-checks) in our setup guide.


<!-- ghprcomment
Source Code Tests
Unit tests – jabsrv
-->

InAnYan commented:
Done

calixtus commented:
Lets move on here. If something comes up or we learn something new, this can always be revised and updated.

calixtus commented:
Merging doc without queue
